infra.bs

```diff
@@ -542,6 +543,74 @@ actually ends up representing <a lt="JavaScript string">JavaScript</a> and
 <a>scalar value strings</a>. It is even fairly typical for implementations to have multiple
 implementations of just <a>JavaScript strings</a> for performance and memory reasons.
 
+<hr>
+
+<p>A <a>string</a> <var>a</var> is a
+<dfn export lt="code unit prefix|starts with">code unit prefix</dfn> of a <a>string</a> <var>b</var>
+if the following steps return true:
+
+<ol>
+ <li><p>Let <var>i</var> be 0.
+
+ <li>
+  <p><a>While</a> true:
+
+  <ol>
+   <li><p>Let <var>aCodeUnit</var> be the <var>i</var>th <a>code unit</a> of <var>a</var>, or null
+   if <var>i</var> is greater than or equal to <var>a</var>'s <a for=string>length</a>.
+
+   <li><p>Let <var>bCodeUnit</var> be the <var>i</var>th <a>code unit</a> of <var>b</var>, or null
+   if <var>i</var> is greater than or equal to <var>b</var>'s <a for=string>length</a>.
+
+   <li><p>If both <var>aCodeUnit</var> and <var>bCodeUnit</var> are null, then return true.
+
+   <li><p>If <var>aCodeUnit</var> is null and <var>bCodeUnit</var> is non-null, then return
+   true.
+
+   <li><p>If <var>aCodeUnit</var> is non-null and <var>bCodeUnit</var> is null, then return
+   false.
+
+   <li><p>Return false if <var>aCodeUnit</var> is different from <var>bCodeUnit</var>.
+
+   <li><p>Set <var>i</var> to <var>i</var> + 1.
```

Comment:
I've used "Increment i by x" for this, but maybe this is clearer? Another thing to settle when we do numbers.

Reply:
"Increment" doesn't appear in Infra yet, so I'll leave it. But we do have an outstanding feature request, in #99.